### ISSUE TITLE
[wishlist] Run auxiliary processes with NO_NEW_PRIVS

### ISSUE THREAD

**Issue by simondeziel** [lxc/lxd#9952]
LXD executes a handful of other processes (dnsmasq, forkdns, tar, gzip, etc) which could possibly be run with the [NO_NEW_PRIVS](https://www.kernel.org/doc/html/latest/userspace-api/no_new_privs.html) flag for an additional security layer.

**Comment by stgraber** [comment]
The first thing to check is whether os.Cmd supports setting no-new-privs, if not, then that's going to be needlessly hard to do.

**Comment by stgraber** [comment]
Doesn't look like it does: https://pkg.go.dev/syscall#SysProcAttr

**Comment by simondeziel** [comment]
https://github.com/opencontainers/runc/blob/main/exec.go#L244 made me think it was possible.

**Comment by tomponline** [comment]
Isn't `AmbientCaps` for this?

**Comment by stgraber** [comment]
No, AmbientCaps is to restrict filesystem/os capabilities instead, it's different from no-new-privs.

**Comment by stgraber** [comment]
Closing this as unfortunately there is no way that I can see to implement this cleanly and I don't think it's worth the hassle and extra cost and security risk of us adding another `forkXYZ` C wrapper just to call the prctl.

We'd need someone to contribute a flag to SysProcAttr upstream and then wait until it's widely available (enough Go versions) before we can make use of this.
